packages/medusa/src/services/product.ts

```diff
@@ -147,7 +156,26 @@ class ProductService extends TransactionBaseService {
       Product & ProductFilterOptions
     >
 
-    return await productRepo.findAndCount(query, q)
+    return await productRepo.findAndCount(query, q)*/
+
+    /**
+     * TODO: The below code is a temporary fix for the issue with the typeorm idle transaction in query strategy mode
+     */
+
+    const manager = this.manager_
```

Comment:
todo: Use active manager
```suggestion
    const manager = this.activeManager_
```